### DISCUSSION TITLE
Is the function to calculate 'EffectiveStrain' in 'RankTwoScalarTools' invalid now?

### DISCUSSION THREAD

**Discussion by echo1115** [idaholab/moose#20979] [Q&A Modules: Solid mechanics] [answered]
According to [https://mooseframework.inl.gov/source/utils/RankTwoScalarTools.html](url) , the effective plastic strain or effective creep strain could be obtained by using the ' **effectiveStrain**' method.
![1651890225(1)](https://user-images.githubusercontent.com/69078335/167234231-91813b6e-eaad-4a3d-82b3-8bc3a806d858.png)

But acturally in MOOSE:
[https://github.com/idaholab/moose/blob/25abc3b0f580842db2924dcdb643bbf4b99406d1/modules/tensor_mechanics/include/utils/RankTwoScalarTools.h](url) , 

the option of  **' effectiveStrain'**  corresponds to a **'mooseError'**.

![1651890129(1)](https://user-images.githubusercontent.com/69078335/167234160-06d102bf-dcc2-4898-86c5-487599f09503.png)

**Comment by hugary1995** [comment, marked as answer]
The error message is accurate, which is also consistent with the documentation.

**Reply by echo1115** [reply]
Does that means  if we want to get the effective plastic strain or effective creep strain，other method should be adopted or we would get an error ?

**Reply by GiudGiud** [reply]
yes you want to be using the new API that the error code is mentioning.
Can you not update to use that one?

**Reply by echo1115** [reply]
yes i can create a new 'Auxkernel' and use "RankTwoScalarTools::effectiveStrain()" to get the effective strain. Thanks for your reply @hugary1995  @GiudGiud
